src/browser/BrowserService.cpp

```diff
@@ -375,22 +375,33 @@ QList<Entry*> BrowserService::searchEntries(Database* db, const QString& hostnam
     }
 
     for (Entry* entry : EntrySearcher().search(hostname, rootGroup, Qt::CaseInsensitive)) {
-        QString title = entry->title();
-        QString url = entry->url();
+        QString entryTitle = entry->title();
+        QString entryUrl = entry->url();
+        QUrl entryQUrl(entryUrl);
+        QString entryScheme = entryQUrl.scheme();
+        int entryPort = entryQUrl.port();
+        QUrl qUrl(url);
+        QString scheme = qUrl.scheme();
+        int port = qUrl.port();
+
+        // Ignore entry if port or scheme defined in the URL doesn't match    
+        if ((entryPort > 0 && entryPort != port) || entryScheme.compare(scheme) != 0) {
+            continue;
+        }
 
         // Filter to match hostname in Title and Url fields
-        if ((!title.isEmpty() && hostname.contains(title))
-            || (!url.isEmpty() && hostname.contains(url))
-            || (matchUrlScheme(title) && hostname.endsWith(QUrl(title).host()))
-            || (matchUrlScheme(url) && hostname.endsWith(QUrl(url).host())) ) {
+        if ((!entryTitle.isEmpty() && hostname.contains(entryTitle))
+            || (!entryUrl.isEmpty() && hostname.contains(entryUrl))
+            || (matchUrlScheme(entryTitle) && hostname.endsWith(QUrl(entryTitle).host()))
+            || (matchUrlScheme(entryUrl) && hostname.endsWith(QUrl(entryUrl).host())) ) {
```

Comment:
Why not just use "entryQUrl" that you defined above? This conditional is next to impossible to decipher.

Reply:
Use it for what? I'm not sure what you mean.

Reply:
You recreate entryQUrl within the conditional when you do QUrl(entryUrl)

Reply:
Ah yes..